modules/videoModule/index.js

```diff
@@ -84,11 +85,26 @@ export function PbVideo(videoCore_, getConfig_, pbGlobal_, pbEvents_, videoEvent
     });
   }
 
-  function renderBid(divId, bid, options = {}) {
-    const adUrl = bid.vastUrl;
-    options.adXml = bid.vastXml;
-    options.winner = bid.bidder;
-    loadAdTag(adUrl, divId, options);
+  async function renderBid(divId, bid, vastUrl, options) {
+    try {
+      const response = await fetch(vastUrl);
+      if (!response.ok) {
+        throw new Error('VAST fetch failed');
+      }
+
+      let vastXml = await response.text();
+
+      // If the fetched VAST XML is a GAM wrapper and local cached is being used,
+      // retrieve the bid's VAST blob from the local cache
+      // and replace the VastAdTagURI with its content.
+      if (config.getConfig('cache.useLocal') && vastXml.includes(bid.videoCacheKey)) {
```

Comment:
This logic is duplicated with what's in the dfp module. You can only get here from `renderWinningBid`, so see below.